src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -693,6 +713,16 @@ public TResult[] ToArray()
                 }
                 return results;
             }
+
+            public List<TResult> ToList()
+            {
+                var results = new List<TResult>(_source.Count);
+                for (int i = 0; i < _source.Count; i++)
```

Comment:
Count here is an interface call.  I'd suggest storing the result into a local and then using the local.